### PR TITLE
Attempt to fix multierror typing

### PR DESCRIPTION
**Pull request by CoolCat467**
At the moment, this is only ignoring generics not matching properly in the subclass. If anyone has any better ideas on how to handle this, feel free to try it out.

### CODE REVIEW

**Comment by TeamSpen210** [comment]
I think the correct solution is to change `MultiError`/`NonBaseMultiError` to also be generic, and propagate that throughout the code.

**Comment by A5rocks** [comment]
I tried and it was annoying (some branch in my fork...)

**Comment by CoolCat467** [comment]
> I think the correct solution is to change `MultiError`/`NonBaseMultiError` to also be generic, and propagate that throughout the code.

That sounds like a good idea, I'll start trying to do that now.

**Comment by TeamSpen210** [comment]
It probably is fine if we add more ignores here, since it's deprecated anyway. Users should only be annotating/interacting with `ExceptionGroup`.

**Comment by CoolCat467** [comment]
It's looking like it might not be possible to make `NonBaseMultiError` generic, because `ExceptionGroup` which we are subclassing itself needs to be using the generic we are defining

**Comment by CoolCat467** [comment]
I've given up attempting to make `NonBaseMultiError` generic, I don't think it's easily possible without doing some nasty things at runtime with monkeypatching

File: trio/_core/_run.py [hunk 1]
```diff
@@ -1705,13 +1701,17 @@ def spawn_impl(
             assert self.init_task is None
 
         ######
-        # Propagate contextvars
+        # Propagate contextvars, and make sure that async_fn can use sniffio.
```
Comment (A5rocks) [inline trio/_core/_run.py:1704]: Err, maybe a bad merge happened or something?

**Review by A5rocks — APPROVED** [review]
Assuming this passes type checking and that the weird previous changes showing in the diff thing is fixed, this looks good to me.

**Comment by A5rocks** [comment]
Actually the merge looks quite messed up -- maybe reset to a clean state, rebase on master, and copy over the type ignore deletion + new type ignore?

File: newsfragments/2700.misc.rst
Comment (A5rocks) [inline newsfragments/2700.misc.rst:1]: Uhhhhh... I think you might have messed up the merge even more 😰

File: trio/_core/_run.py (hunk 1 above)
Comment (CoolCat467) [reply trio/_core/_run.py]: Yes, very bad, but it's fixed now.

**Review by A5rocks — APPROVED** [review]
Looks good

**Pull request merged**